checknix.py

```diff
@@ -0,0 +1,64 @@
+import os, sys, shutil
+
+from textwrap import dedent
+import tempfile
+
+import distutils.sysconfig
+import distutils.ccompiler
+from distutils.errors import CompileError, LinkError
+
+
+def check_nix(libdirs=(), incdirs=()):
+    """
+    Check if NIX is available by trying to compile a tiny program.
+    """
+
+    cpp_code = dedent("""
+    #include <nix.hpp>
+
+    int main() {
+        return 0;
+    }
+    """)
+
+    tmpdir = tempfile.mkdtemp()
+    bin_file_name = os.path.join(tmpdir, "check_nix")
+    file_name = bin_file_name + '.cpp'
+    with open(file_name, 'w') as cfile:
+        cfile.write(cpp_code)
+
+    compiler = distutils.ccompiler.new_compiler()
+    assert isinstance(compiler, distutils.ccompiler.CCompiler)
+
+    compiler.library_dirs.extend(libdirs)
+    compiler.include_dirs.extend(incdirs)
+    distutils.sysconfig.customize_compiler(compiler)
+
+    stderr = os.dup(sys.stderr.fileno())
+    errfile = open(os.path.join(tmpdir, "check_nix.err"), 'w')
+    # redirect stderr to file
+    os.dup2(errfile.fileno(), sys.stderr.fileno())
+    try:
+        compiler.compile([file_name], output_dir=tmpdir,
+                         extra_postargs=["--std=c++11"])
```

Comment:
NIX (headers) also depend on boost (headers) so if boost is installed in a non-standard location then nix will fail to build. Could work around that by also passing the boost incdirs.